### DISCUSSION TITLE
CodeLens samples and/or guideline for VS2022

### DISCUSSION THREAD

**Discussion by PongNT** [VsixCommunity/Community.VisualStudio.Toolkit#292] [Q&A]
Hello guys, 

I am desperately looking for a good tutorial, official documentation or canonical example to develop an extension with a codelens interface. Where can I find this?  I found some examples but either they are not up to date (it seems there are significant changes with VS2022) or they are different from each other and finally I don't know what is the proper way to implement codelens. 

Thank you for your help.

**Comment by reduckted** [comment]
https://github.com/microsoft/VSSDK-Extensibility-Samples/tree/master/CodeLensOopSample

That sample worked for me in Visual Studio 2022 with one change. I had to add the `CodeLensOopProvider` as a `MefComponent` to the `CodeLensOopProviderVsix` project. It's currently listed as a `CodeLensComponent`, but that doesn't show up in the dropdown of asset types, and the assembly wasn't being loaded until I made it a `MefComponent`.
